### PR TITLE
fix(deps): update raw-window-handle to v0.6

### LINKED ISSUE #1143: Update raw-window-handle dependency to v0.6
**Is your feature request related to a problem? Please describe.**

raw-window-handle crate dependency of this crate is out of date. It is better to update it to the latest because

- some crates require the latest version of raw-window-handle. For example, wgpu v0.19 requires raw-window-handle v0.6 and doesn't work with v0.5
- v0.6 provides safer interface. `WindowHandle` manages lifetime not to make the underlying handle invalid. It is better to make users migrate to the safer interface
- Currently wry has both v0.6 and v0.5 as dev dependency redundantly because winit enables `rwh_06` by default

**Describe the solution you'd like**

Bump up raw-window-handle to v0.6. This means the following breaking changes:

- wry's APIs will require `HasWindowHandle` trait instead of `HasRawWindowHandle` since `HasRawWindowHandle` was deprecated at v0.6
- `wry::raw_window_handle` will re-export v0.6

**Describe alternatives you've considered**

- Support both v0.6 and v0.5 as winit does. This would increase maintenance cost but possible
- Keep v0.5 until v0.6 is mandatory due to some reason

**Additional context**

Nothing.

### PR DESCRIPTION
**Pull request by rhysd**
Fix #1143
Close #1062 

I confirmed

- `cargo check`
- `cargo test`
- `winit`, `simple`, `wgpu` examples

on the following targets

- `x86_64-apple-darwin`
- `x86_64-pc-windows-msvc`
- `x86_64-unknown-linux-gnu`

And I confirmed `cargo check` on the following targets (because I don't have the environments)

- `aarch64-apple-darwin`
- `x86_64-linux-android`
- `aarch64-apple-ios`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -117,6 +117,6 @@ tao-macros = { version = "0.1.0" }
 [dev-dependencies]
 http-range = "0.1.5"
 pollster = "0.3.0"
-tao = { version = "0.24", default-features = false, features = [ "rwh_05" ] }
-wgpu = "0.18.0"
-winit = { version = "0.29", features = [ "rwh_05" ] }
+tao = { version = "0.24", default-features = false, features = [ "rwh_06" ] }
```
Comment (rhysd) [inline Cargo.toml]: I was not sure this line should be simplified to `tao = "0.24"` so I preserved the configuration as much as possible.

Comment (amrbashir) [reply Cargo.toml]: let's simplify it

Comment (rhysd) [reply Cargo.toml]: Sure.

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Thanks

**Pull request merged**